pkg/bootflow/trustchains/trust_chain_tpm.go

```diff
@@ -0,0 +1,108 @@
+package trustchains
+
+import (
+	"fmt"
+
+	"github.com/9elements/converged-security-suite/v2/pkg/bootflow/types"
+	pcrtypes "github.com/9elements/converged-security-suite/v2/pkg/pcr/types"
+	"github.com/google/go-tpm/tpm2"
+)
+
+var _ types.TrustChain = (*TPM)(nil)
+
+type TPM struct {
+	PCRValues PCRValues
+	EventLog  TPMEventLog
+}
+
+func TPMFunc(state *types.State, fn func(tpm *TPM) error) error {
+	for _, trustChain := range state.TrustChains {
+		if tpm, ok := trustChain.(*TPM); ok {
```

Comment:
the fact that you need to check rt type info here seems like a bit of code smell to me. you could probably keep a map of known chains and avoid this func completely; it wouldnt avoid extensibility, although realistically speaking, it's not every day that a new trust mechanism is added

Reply:
i see the same pattern happens with state.SystemArtifacts, and you need to do a linear search every time there as well

Reply:
Fixed. Now we use a map instead